examples/webserver/webserver.py

```diff
@@ -72,28 +72,36 @@ async def send_file(relative_path, request_headers):
     if 'X-Real-IP' in request_headers:
         print(f'GET {relative_path} {request_headers["X-Real-IP"]}')
 
-    if '..' in relative_path:
+    if '..' in relative_path.split('/'):
         # directory traversal is not allowed
         return Response(status_code=404, reason='Not Found')
-    absolute_path = f'{base_dir}/{relative_path}'
-
-    if absolute_path.endswith('/'):
-        absolute_path = f'{absolute_path}index.html'
-
-    try:
-        stat = await fs_async.stat(absolute_path)
-    except:
+    absolute_path = f'{base_dir}/{relative_path}'.rstrip('/')
```

Comment:
We have had cases before where `http://example.com/fred.html` (accessed via `http://example.com/fred`and `http://example.com/fred/` are different URLs.
Please don't strip the trailing slash.
I suggest instead, 
```
if relative_path.endswith('/'):
    relative_path = relative_path + "index"
suffices = [ '', '.html', '.htm', '.xhtml' ]
for suffix in suffices:
  try stat = await fs_async.stat(relative_path + suffix)
  ...
  ```

This avoids pre-computing lots of paths, and creates only strings that are actually used.
If there's a trailing slash, it means this is a directory, and the index file inside that directory is whats wanted.  For our purposes at the moment that's always `index.html` but most webservers allow it to be configured.